### PR TITLE
Add Documentation 1.0.0b

### PR DESCRIPTION
**Pull request by elenatorro**
Related issues:
* https://github.com/CartoDB/cartoframes/issues/750
* https://github.com/CartoDB/cartoframes/issues/755

- [x] Examples: https://github.com/CartoDB/cartoframes/pull/769
- [x] Guides: https://github.com/CartoDB/cartoframes/pull/786
- [x] Support multiple versions in devcenter for CARTOframes https://github.com/CartoDB/developers/issues/595

Developer center WIP: http://test-cartoframes-documentation.developers.carto-staging.com/developers/cartoframes

### Acceptance:

Examples:
- Run examples locally and check it looks 👌 
- Check Page http://test-cartoframes-documentation.developers.carto-staging.com/developers/cartoframes/examples/

Guides:
- Review guides http://test-cartoframes-documentation.developers.carto-staging.com/developers/cartoframes/guides/

### CODE REVIEW

File: docs/developer-center/guides/01-Quickstart.md [hunk 1]
```diff
@@ -55,98 +60,138 @@ Next, create a new notebook. See Jupyter's [running a notebook](https://jupyter.
...
-```python
-cc.write(df, 'brooklyn_poverty', overwrite=True)
+```py
+from cartoframes.viz import Map, Layer, Legend
+
+Map(
+    Layer(
+        'spend_data',
+        'color: ramp($category, bold)',
+        legend=Legend({
```
Comment (Jesus89) [inline docs/developer-center/guides/01-Quickstart.md]: Remove Legend. Use { 'type': 'color-bins', 'title': 'Categories' } instead

File: docs/developer-center/guides/01-Quickstart.md [hunk 2]
```diff
@@ -55,98 +60,138 @@ Next, create a new notebook. See Jupyter's [running a notebook](https://jupyter.
...
-cc.map()
+Map(
+    Layer(
+        'spend_data',
+        'color: ramp($category, bold)',
+        legend=Legend({
+            'type': 'color-bins',
+            'title': 'Categories'
+        }),
+        popup=Popup({
```
Comment (Jesus89) [inline docs/developer-center/guides/01-Quickstart.md]: Remove Legend and Popup classes

File: docs/developer-center/guides/02-Context.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+The elements needed to create contexts are under the `cartoframes.auth` namespace.
+
+### Default Context
+
+With `set_default_context`, the same context will be used by all the layers and sources by default.
+
+```py
+from cartoframes.auth import set_default_context
+
+set_default_context(
```
Comment (Jesus89) [inline docs/developer-center/guides/02-Context.md]: `set_default_context` allows this `set_default_context(https://your_user_name.carto.com/)` for public datasets. I would comment this in the guides

File: docs/developer-center/guides/02-Context.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+You can set a `Context` for a `Layer`, and different contexts for different leyers:
+
+```py
+from cartoframes.auth import Context
+from cartoframes.viz import Map, Layer
+
+visualization = Map(
+    Layer(
+        'dataset',
+        context=Context('https://your_user_name.carto.com/', 'default_public')
```
Comment (Jesus89) [inline docs/developer-center/guides/02-Context.md]: Context class has also `default_public` as default API key

File: docs/developer-center/guides/04-Sources.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+```py
+from cartoframes.viz import Map, Layer
+from cartoframes.data import Dataset
+
+ds = Dataset.from_table('populated_places')
+
+Map(Layer(ds))
+```
+
+### Pandas Dataframe
```
Comment (Jesus89) [inline docs/developer-center/guides/04-Sources.md]: This final example was added in the debug notebooks just to test the Dataset/Dataframe scenario, but IMO we could remove it here.

File: docs/developer-center/guides/05-Legends.md [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+* color-category-line
+* color-category-polygon
+
+Example:
+```py
+Map(
+    Layer(
+        'populated_places',
+        'color: ramp(top($adm0name, 5), bold)',
+        legend=Legend({
```
Comment (Jesus89) [inline docs/developer-center/guides/05-Legends.md]: Remove Legend class

File: docs/developer-center/guides/06-Popups.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+![Set Popup for 'hover' events using expressions](../../img/guides/popups/popups-2.png)
+
+### Adding titles
+
+```py
+Map(
+    Layer(
+        'sf_neighborhoods',
+        popup=Popup({
```
Comment (Jesus89) [inline docs/developer-center/guides/06-Popups.md]: Remove Popup class

File: docs/developer-center/guides/08-Publishing.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+Finally, we will be ready to publish the visualization!
+
+```py
+from cartoframes.viz import Map, Layer
+from cartoframes.data import Dataset
+
+ds = Dataset.from_table('private_table_name')
+ds.download()
+ds._is_saved_in_carto = False
```
Comment (Jesus89) [inline docs/developer-center/guides/08-Publishing.md]: I think this is a workaround

File: examples/_debug/basemaps.ipynb [hunk 9]
```diff
@@ -792,7 +792,11 @@
        "</iframe>"
       ],
       "text/plain": [
+<<<<<<< HEAD:examples/_debug/basemaps.ipynb
```
Comment (Jesus89) [inline examples/_debug/basemaps.ipynb]: Merge conflict here

File: examples/_debug/legend-color-category.ipynb [hunk 10]
```diff
@@ -800,10 +798,14 @@
        "</iframe>"
       ],
       "text/plain": [
+<<<<<<< HEAD:examples/_debug/legend-color-category.ipynb
```
Comment (Jesus89) [inline examples/_debug/legend-color-category.ipynb]: Merge conflict here

File: examples/debug/API/source-geojson.ipynb [hunk 11]
```diff
@@ -1,4065 +0,0 @@
-{
```
Comment (Jesus89) [inline examples/debug/API/source-geojson.ipynb]: It would be nice to keep this debug notebook

**Comment by makella** [comment]
@elenatorro awesome! taking a look through as well.

**Comment by makella** [comment]
hi @elenatorro!

As a first pass, this is a lot of great information, nice job on getting everything together like you have!

For this review, I did a quick skim of each guide and mostly did some proofreading changes.

I have some other thoughts/ideas, but since we will continue to work on these, I'll save those for later!

One comment is about the Quickstart guide. I started going through it but realized that our initial idea of making the flow similar to the existing Quickstart actually might not be needed for the new version since we have detailed guides about the other pieces (legends, helpers, popups, etc.). What we might want to do is end the Quickstart after installation and authentication since the other guides cover the other parts. Let's definitely discuss this later.

For the quickstart, I did proofread edits also, We can go through content and structure later this week.

Awesome job with all of this!

File: examples/debug/API/source-geojson.ipynb (hunk 11 above)
Comment (elenatorro) [reply examples/debug/API/source-geojson.ipynb]: It has the same content as the geojson public example, if we need a debug example later to try more things we can add it then 👌

Comment (Jesus89) [reply examples/debug/API/source-geojson.ipynb]: :+1:

File: examples/_debug/basemaps.ipynb [hunk 12]
```diff
@@ -1584,7 +1584,11 @@
...
```
Comment (Jesus89) [inline examples/_debug/basemaps.ipynb]: Merge conflict here

**Comment by makella** [comment]
Nice!!

**Pull request merged**